### PR TITLE
Code Review for Messaging refactor develop branch

### PR DESCRIPTION
**Pull request by mgkwill**
### **NOT MEANT FOR MERGE!!**
Objective of pull request: Code review vehicle for messaging_refactor_develop branch compared to main.

### CODE REVIEW

**Comment by killight98** [comment]
Updats for defining APIs of msp passing lib (Py & Cpp)
1. Add relative Py APIs into message_infrastructure package
```
magma/runtime/message_infrastructure/multiprocessing.py -> message_infrastructure/multiprocessing.py
magma/core/model/py/ports.py -> message_infrastructure/ports.py
```
2. Integrate `Channel`, `RecvPort` and `SendPort` APIs into LAVA src
3. Lint Code are passed including cpplint.
4. Installing LAVA is successful including message_infrastructure package.

Some interface parameters (input/output) are not complete and we will implement it at next phase.

File: src/lava/magma/compiler/builders/py_builder.py [hunk 1]
```diff
@@ -7,15 +7,18 @@
...
+    SendPort
+)
 from lava.magma.compiler.utils import (PortInitializer, VarInitializer,
                                        VarPortInitializer)
 from lava.magma.core.model.py.model import AbstractPyProcessModel
-from lava.magma.core.model.py.ports import (AbstractPyIOPort,
-                                            IdentityTransformer, PyInPort,
-                                            PyOutPort, PyRefPort, PyVarPort,
-                                            VirtualPortTransformer)
+from message_infrastructure.ports import (AbstractPyIOPort,
```
Comment (joyeshmishra) [inline src/lava/magma/compiler/builders/py_builder.py:18]: Is it possible to bring it under the lava.message_infrastructure umbrella? Please check this with @awintel regarding convention.

**Review by joyeshmishra — APPROVED** [review]
Looks good. Not sure how the message_infrastructure naming should look like. Please discuss.

File: src/lava/magma/compiler/builders/py_builder.py (hunk 1 above)
Comment (killight98) [reply src/lava/magma/compiler/builders/py_builder.py:18]: Yes. Need to do following steps:
- move the code from `src/lava/magma/runtime/message_infrastructure` to `src/lava/message_infrastructure`.
- remove the sub-project message_infrastructure and build cpp message_infrastructure under lava potery.

File: src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/abstract_actor.h [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+#define ABSTRACT_ACTOR_H_
+
+#include <functional>
+
+namespace message_infrastructure {
+
+enum ActorStatus {
+  StatsError = -1,
+  StatsRuning = 0,
+  StatsStopped = 1
```
Comment (ysingh7) [inline src/lava/magma/runtime/message_infrastructure/message_infrastructure/csrc/abstract_actor.h:15]: I think there is a typo. Status?

**Review by ysingh7 — COMMENTED** [review]
I think overall the code looks really good. One thing which is missing consistently is lack of any docstring or comments. Documenting what the class does and its function will improve the overall quality of code even more.

File: src/lava/magma/compiler/builders/runtimeservice_builder.py [hunk 3]
```diff
@@ -115,21 +119,21 @@ def build(self) -> AbstractRuntimeService:
 
         if not nxsdk_rts:
             for port in self.csp_proc_send_port.values():
-                if "service_to_process" in port.name:
+                if "service_to_process" in port.name():
                     rs.service_to_process.append(port)
```
[the remark this answers is not in the document]
Comment (hongdami) [reply src/lava/magma/compiler/builders/runtimeservice_builder.py:123]: I will change the name from funtion to attribute

File: src/lava/magma/compiler/builders/runtimeservice_builder.py [hunk 4]
```diff
@@ -68,12 +72,12 @@ def set_csp_ports(self, csp_ports: ty.List[AbstractCspPort]):
...
-                self.csp_send_port.update({port.name: port})
-            if isinstance(port, CspRecvPort):
-                self.csp_recv_port.update({port.name: port})
+            if isinstance(port, SendPort):
+                self.csp_send_port.update({port.name(): port})
+            if isinstance(port, RecvPort):
+                self.csp_recv_port.update({port.name(): port})
 
-    def set_csp_proc_ports(self, csp_ports: ty.List[AbstractCspPort]):
+    def set_csp_proc_ports(self, csp_ports: ty.List[AbstractTransferPort]):
```
[the remark this answers is not in the document]
Comment (hongdami) [reply src/lava/magma/compiler/builders/runtimeservice_builder.py:80]: We create the AbstractTrasferPort to wrapper all the channel port type in cpplib which includes shmport, rpcport  ...
Do you have any better advice on it?

File: src/lava/magma/compiler/compiler.py [hunk 5]
```diff
@@ -728,7 +730,7 @@ def _create_sync_channel_builders(
         sync_channel_builders: ty.List[AbstractChannelBuilder] = []
         for sync_domain in rsb:
             runtime_to_service = RuntimeChannelBuilderMp(
-                ChannelType.PyPy,
+                ChannelTransferType.SHMEMCHANNEL,
```
[the remark this answers is not in the document]
Comment (hongdami) [reply src/lava/magma/compiler/compiler.py:733]: sure

File: src/lava/magma/compiler/builders/runtimeservice_builder.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (ysingh7) [reply src/lava/magma/compiler/builders/runtimeservice_builder.py:80]: I think its fine

**Pull request closed without merging**